pkg/trace/obfuscate/sql.go

```diff
@@ -151,22 +152,58 @@ func (f *groupingFilter) Reset() {
 // Process the given SQL or No-SQL string so that the resulting one is properly altered. This
 // function is generic and the behavior changes according to chosen tokenFilter implementations.
 // The process calls all filters inside the []tokenFilter.
-func obfuscateSQLString(in string) (string, error) {
-	tokenizer := NewSQLTokenizer(in)
+func (o *Obfuscator) obfuscateSQLString(in string) (string, error) {
+	ignoreEscapes := o.ShouldIgnoreEscapes()
+	tokenizer := NewSQLTokenizer(in, ignoreEscapes)
 	filters := []tokenFilter{&discardFilter{}, &replaceFilter{}, &groupingFilter{}}
+
+	out, err := attemptObfuscation(tokenizer, filters)
+	if err != nil {
```

Comment:
We could just:
```go
if err != nil && tokenizer.SeenEscape() {
    // we might've failed because of taking the wrong decision about handling
    // escape characters; try again differently.
```